cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+   * @brief Copies the underying gdf_column's data and bitmask to the host.
+   *
+   * Returns a tuple of two std::vectors. The first is the column's data, and
+   * the second is the column's bitmask.
+   *
+   *---------------------------------------------------------------------------**/
+  auto to_host() const {
+    gdf_size_type const num_masks{gdf_get_num_chars_bitmask(the_column.size)};
+    std::vector<ColumnType> host_data;
+    std::vector<gdf_valid_type> host_bitmask;
+
+    if (nullptr != the_column.data) {
+      // TODO Is there a nicer way to get a `std::vector` from a device_vector?
+      host_data.resize(the_column.size);
+      CUDA_RT_CALL(cudaMemcpy(host_data.data(), the_column.data,
+                              the_column.size * sizeof(ColumnType),
+                              cudaMemcpyDeviceToHost));
+    }
+
+    if (nullptr != the_column.valid) {
+      host_bitmask.resize(num_masks);
+      CUDA_RT_CALL(cudaMemcpy(host_bitmask.data(), the_column.valid,
+                              num_masks * sizeof(gdf_valid_type),
+                              cudaMemcpyDeviceToHost));
+    }
+
+    return std::make_tuple(host_data, host_bitmask);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the values of the underlying gdf_column.
+   *
+   *---------------------------------------------------------------------------**/
+  void print() const {
+    // TODO Move the implementation of `print_gdf_column` here once it's removed
+    // from usage elsewhere
+    print_gdf_column(&the_column);
+  }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Functor for comparing if two elements between two gdf_columns are
+   * equal.
+   *
+   *---------------------------------------------------------------------------**/
+  struct elements_equal {
+    gdf_column lhs_col;
+    gdf_column rhs_col;
+    bool nulls_are_equivalent;
+
+    /**---------------------------------------------------------------------------*
+     * @brief Constructs functor for comparing elements between two gdf_column's
+     *
+     * @param lhs The left column for comparison
+     * @param rhs The right column for comparison
+     * @param nulls_are_equal Desired behavior for whether or not nulls are
+     * treated as equal to other nulls. Defaults to true.
+     *---------------------------------------------------------------------------**/
+    __host__ __device__ elements_equal(gdf_column lhs, gdf_column rhs,
+                                       bool nulls_are_equal = true)
+        : lhs_col{lhs}, rhs_col{rhs}, nulls_are_equivalent{nulls_are_equal} {}
+
+    __device__ bool operator()(gdf_index_type row) {
+      bool const lhs_is_valid{gdf_is_valid(lhs_col.valid, row)};
+      bool const rhs_is_valid{gdf_is_valid(rhs_col.valid, row)};
+
+      if (lhs_is_valid && rhs_is_valid) {
```

Comment:
If you're using the human-readable keywords, why not say `lhs_is_valid and rhs_is_valid`?

Reply:
Using the human readable keywords is a habit I'm still trying to adopt and sometimes I forget.